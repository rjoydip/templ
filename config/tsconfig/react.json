{
  "compilerOptions": {
    "target": "ES2020",
    "jsx": "react",
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "useDefineForClassFields": true,

    "baseUrl": ".",
    "module": "ESNext",

    /* Bundler mode */
    "moduleResolution": "bundler",
    "paths": {
      "@/*": [
        "./src/*"
      ]
    },
    "resolveJsonModule": true,

    /* Linting */
    "strict": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noEmit": true,
    "isolatedModules": true,
    "skipLibCheck": true
  },
  "references": [{ "path": "./base.json" }],
  "include": ["src"]
}
