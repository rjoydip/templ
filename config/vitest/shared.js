const configShared = {
  test: {
    include: ['{test,tests}/**/*.test.{ts,js}'],
    coverage: {
      enabled: true,
      reporter: ['text', 'json', 'html'],
    },
  },
}

export default configShared
