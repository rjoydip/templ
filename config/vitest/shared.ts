const configShared = {
  test: {
    include: ['{test,tests}/**/*.test.{ts,js}'],
    exclude: ['esbuild.config.ts*'],
    coverage: {
      enabled: true,
      reporter: ['text', 'json', 'html'],
    },
  },
}

export default configShared
