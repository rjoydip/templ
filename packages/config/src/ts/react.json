{
  "display": "React",
  "extends": "./base.json",
  "compilerOptions": {
    "module": "ES2022",
    "lib": ["ES2022", "DOM", "DOM.Iterable"],
    "moduleResolution": "bundler",
    "jsx": "react"
  }
}
