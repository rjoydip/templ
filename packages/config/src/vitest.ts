export function getVitestConfig() {
  return {
    test: {
      include: ['test/**/*.test.ts'],
      coverage: {
        enabled: true,
        reporter: ['text', 'json', 'html'],
      },
    },
  }
}
