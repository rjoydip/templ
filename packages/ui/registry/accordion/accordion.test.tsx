import { cleanup, fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'
import {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
} from './accordion'

function renderAccordion({ args }: any) {
  const { container } = render(
    <Accordion {...args}>
      <AccordionItem value="item-1">
        <AccordionTrigger>Is it accessible?</AccordionTrigger>
        <AccordionContent>
          Yes. It adheres to the WAI-ARIA design pattern.
        </AccordionContent>
      </AccordionItem>
      <AccordionItem value="item-2">
        <AccordionTrigger>Is it styled?</AccordionTrigger>
        <AccordionContent>
          Yes. It comes with default styles that matches the other components'
          aesthetic.
        </AccordionContent>
      </AccordionItem>
      <AccordionItem value="item-3">
        <AccordionTrigger>Is it animated?</AccordionTrigger>
        <AccordionContent>
          Yes. It's animated by default, but you can disable it if you prefer.
        </AccordionContent>
      </AccordionItem>
    </Accordion>,
  )
  return container
}

const { getByText } = screen

describe('accordion component', () => {
  let container: HTMLElement

  beforeEach(() => {
    container = renderAccordion({})
  })

  afterEach(() => cleanup())

  it('should render properly', () => {
    expect(container).toMatchSnapshot()
  })
  it('should validate a11y violations for accordion', async () => {
    expect(await axe(container)).toHaveNoViolations()
  })

  it('should render content when accordion element is clicked', async () => {
    const ele = getByText(/Is it accessible?/)
    expect(ele).toBeDefined()
    fireEvent.click(ele)
  })
})
