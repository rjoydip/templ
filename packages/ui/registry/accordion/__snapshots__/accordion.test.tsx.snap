// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`accordion component > should render properly 1`] = `
<div>
  <div
    data-orientation="vertical"
  >
    <div
      class="border-b"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          class="flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          Is it accessible?
          <svg
            class="h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r0:"
        class="overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r1:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="border-b"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          class="flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r2:"
          type="button"
        >
          Is it styled?
          <svg
            class="h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="border-b"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r5:"
          aria-expanded="false"
          class="flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r4:"
          type="button"
        >
          Is it animated?
          <svg
            class="h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r4:"
        class="overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r5:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
